### ISSUE TITLE
automated parameter optimization ideas

### ISSUE THREAD

**Issue by kushalkolar** [flatironinstitute/CaImAn#1082]
Making an issue to just collect ideas since @EricThomson and I have thought about this every now and then for years.

1. I wonder what the Fourier transforms of the residual mean, A *C mean and b * f mean look like and if they could be used to construct a loss function. It might be useful to use the Earth Mover's Distance for these. I think we could make a cost function to minimize EMD between A * C and input movie, and simultaneously maximize EMD between residuals and input movie.

2. Since FIOLA is very fast, it might be practically possible to perform gradient descent to optimize within the parameter space.
